tests/turnstile_chain.c

```diff
@@ -0,0 +1,106 @@
+#include <ktest.h>
+#include <klog.h>
+#include <mutex.h>
+#include <runq.h>
+#include <sched.h>
+#include <thread.h>
+
+// length of the propagation chain
+#define T 4
+
+// TODO replace asserts about priority borrowing with some accessor
+/* TODO consider where we could check restoring priorities
+ * I think that all threads should keep HIGH priority until the last propagator
+ * has acquired 2nd mutex and then the priorities would be restored to initial
+ * ones (propagator_priority(i) for propagator[i]).
+ */
+
+static mtx_t mtx[T + 1];
+// TODO consider whether this would be any useful
+// volatile int mtx_passed[T + 1];
+
+// for simpler code: propagator[0] = starter
+static thread_t *propagator[T + 1];
+static thread_t *starter;
+
+static void set_base_prio(thread_t *td, prio_t prio) {
+  WITH_SPINLOCK(td->td_spin) {
+    sched_set_prio(td, prio);
+  }
```

Comment:
1) Name `set_base_prio` is misleading as it sets both base and active prio.
2) Remember that `sched_set_prio` will be using `turnstile_adjust` in the future. Maybe we don't want to call it in our turnstile tests.

Reply:
Done.

Reply:
Also, 2.: I don't think we should mind. We will just test a little more functions (although `turnstile_adjust` in this case will perform very simple action because `set_prio` is called only before the thread is added).